imsv-docs-astro/src/components/registry-views/tokens/NetworkTokensTable.astro

```diff
@@ -0,0 +1,54 @@
+---
+import { registry } from 'src/content/registry.mjs';
+import { wrappableAddress } from 'src/components/registry-views/wrappableAddress';
+
+const {
+  netType,
+  token,
+  columns=[ 'symbol', 'chain', 'address', 'faucet' ],
+} = Astro.props;
+
+const networkTokens = registry.allNetworkTokens()
+  .filter(networkToken => !netType || networkToken.network.type == netType)
+  .filter(networkToken => !token || networkToken.token.name == token);
+
+const columnTitle = {
+  symbol: 'Symbol',
+  chain: 'Chain',
+  address: 'Token Address',
+  faucet: 'Faucet',
+};
+
+---
+
+<div class="overflow-x-auto">
+  <table>
+    <thead>
+      <tr>
+        {columns.map(column => (
+          <th>{columnTitle[column]}</th>
+        ))}
+      </tr>
+    </thead>
+      <tbody>
+      {networkTokens.map(networkToken => (
+        <tr>
+          {columns.map(column=> {
+            switch(column) {
+              case 'symbol':
+                return <td><nobr>{networkToken.token.title}</nobr></td>
+              case 'chain':
+                return <td><nobr>{networkToken.network.chain.title}</nobr></td>
+              case 'address':
+                return <td><a href={networkToken.url} set:html={wrappableAddress(networkToken.address)}></a></td>
+              case 'faucet':
+                return <td><a href={networkToken.faucetUrl}>{networkToken.faucetTitle}</a></td>
+              default:
+                return <td></td>
```

Comment:
All looks good, only comment would be that maybe these switches are unnecessary, it's nice to map over the columns but doesn't really make the table generation any more dynamic since a new case needs to be added every time a new column is added. Nice to have the cases as labels to see which content belongs to which column though.